lib/matplotlib/axes/_axes.py

```diff
@@ -239,8 +239,8 @@ def legend(self, *args, **kwargs):
             selection by using a label starting with an underscore, "_".
             A string starting with an underscore is the default label for all
             artists, so calling `.Axes.legend` without any arguments and
-            without setting the labels manually will result in no legend being
-            drawn.
+            without setting the labels manually will result in a ``UserWarning``
+            and no legend being drawn.
```

Comment:
I think this isn't (and was not) correct. We draw an empty legend.

Reply:
Happy to, but a bit confused.  When I run the following, no legend appears on the image:

```python
plt.plot([1, 2, 3])
ax.lengend()
```

Is this a backend-bependent behavior?  Or by "draw" do you mean the legend object is created and some drawing code runs on it, even if there's no visible legend?

Reply:
There is actually a tiny gray box in the upper right hand corner when you do that
<img width="201" alt="NoLegend" src="https://github.com/matplotlib/matplotlib/assets/1562854/5791fdbf-57c4-4d4d-91e4-e0f104c32f77">

Reply:
Correct, the legend is empty and thus shrinks to the small box.

Reply:
Ahh, I totally missed that, haha